src/controllers/controllerpresetfilehandler.cpp

```diff
@@ -13,56 +13,53 @@
 #include "controllers/hid/hidcontrollerpresetfilehandler.h"
 
 // static
-ControllerPresetPointer ControllerPresetFileHandler::loadPreset(const QString& pathOrFilename,
-                                                                const QStringList& presetPaths) {
-    qDebug() << "Searching for controller preset" << pathOrFilename
-             << "in paths:" << presetPaths.join(",");
-    QString scriptPath = ControllerManager::getAbsolutePath(pathOrFilename,
-                                                            presetPaths);
-
-    if (scriptPath.isEmpty()) {
-        qDebug() << "Could not find" << pathOrFilename
-                 << "in any preset path.";
+ControllerPresetPointer ControllerPresetFileHandler::loadPreset(
+        const QFileInfo& presetFile, const QDir& systemPresetsPath) {
+    VERIFY_OR_DEBUG_ASSERT(presetFile.exists() && presetFile.isReadable()) {
+        qDebug() << "Preset" << presetFile.absoluteFilePath()
+                 << "does not exist or is unreadable.";
         return ControllerPresetPointer();
     }
 
-    QFileInfo scriptPathInfo(scriptPath);
-    if (!scriptPathInfo.exists() || !scriptPathInfo.isReadable()) {
-        qDebug() << "Preset" << scriptPath << "does not exist or is unreadable.";
-        return ControllerPresetPointer();
-    }
-
-    ControllerPresetFileHandler* pHandler = NULL;
-    if (scriptPath.endsWith(MIDI_PRESET_EXTENSION, Qt::CaseInsensitive)) {
+    ControllerPresetFileHandler* pHandler = nullptr;
+    if (presetFile.fileName().endsWith(
+                MIDI_PRESET_EXTENSION, Qt::CaseInsensitive)) {
         pHandler = new MidiControllerPresetFileHandler();
-    } else if (scriptPath.endsWith(HID_PRESET_EXTENSION, Qt::CaseInsensitive) ||
-               scriptPath.endsWith(BULK_PRESET_EXTENSION, Qt::CaseInsensitive)) {
+    } else if (presetFile.fileName().endsWith(
+                       HID_PRESET_EXTENSION, Qt::CaseInsensitive) ||
+            presetFile.fileName().endsWith(
+                    BULK_PRESET_EXTENSION, Qt::CaseInsensitive)) {
         pHandler = new HidControllerPresetFileHandler();
```

Comment:
It's confusing that this class is called `HidControllerPresetFileHandler` but here it is also being used for bulk controllers.

Reply:
Yes, but I don't want to do too much refactoring tbh.

Reply:
That's fine, we don't have to solve this now.